src/corehost/cli/breadcrumbs.cpp

```diff
@@ -7,36 +7,48 @@
 #include "trace.h"
 #include "breadcrumbs.h"
 
-breadcrumb_writer_t::breadcrumb_writer_t(const std::unordered_set<pal::string_t>* files)
+breadcrumb_writer_t::breadcrumb_writer_t(bool enabled, const std::unordered_set<pal::string_t>* files)
     : m_status(false)
+    , m_enabled(enabled)
     , m_files(files)
 {
-    if (!pal::get_default_breadcrumb_store(&m_breadcrumb_store))
+    if (enabled && !pal::get_default_breadcrumb_store(&m_breadcrumb_store))
     {
         m_breadcrumb_store.clear();
     }
 }
 
+breadcrumb_writer_t::~breadcrumb_writer_t()
+{
+    if (m_enabled)
+    {
+        end_write();
```

Comment:
Do we need to be concerned about an exception happening in here?

I am not an expert on this.

https://akrzemi1.wordpress.com/2011/09/21/destructors-that-throw/

Reply:
There are no expected exceptions in the breadcrumb dtor; the AV that was occurring in end_write should no longer occur.